lib/queue_bus/config.rb

```diff
@@ -1,102 +1,70 @@
-      end
-      @adapter_instance
-    end
-
-    def adapter
-      return @adapter_instance if @adapter_instance
-      raise "no adapter has been set"
-    end
-
-    def redis(&block)
-      # TODO: could allow setting for non-redis adapters
-      adapter.redis(&block)
-    end
+# frozen_string_literal: true
 
-    def default_app_key=val
-      @default_app_key = Application.normalize(val)
-    end
+require 'socket'
+require 'logger'
 
-    def default_app_key
-      @default_app_key
-    end
+module QueueBus
+  # This class contains all the configuration for a running queue bus application.
+  class Config
+    attr_accessor :default_queue, :local_mode, :hostname, :incoming_queue, :logger
 
-    def default_queue=val
-      @default_queue = val
-    end
+    attr_reader :worker_middleware_stack
 
-    def default_queue
-      @default_queue
+    def initialize
+      @worker_middleware_stack = QueueBus::Middleware::Stack.new
+      @incoming_queue = 'bus_incoming'
+      @hostname = Socket.gethostname
     end
 
-    def local_mode=value
-      @local_mode = value
-    end
+    def adapter=(val)
+      raise "Adapter already set to #{@adapter_instance.class.name}" if has_adapter?
 
-    def local_mode
-      @local_mode
+      @adapter_instance =
+        if val.is_a?(Class)
+          val.new
+        elsif val.is_a?(::QueueBus::Adapters::Base)
+          val
+        else
+          class_name = ::QueueBus::Util.classify(val)
+          ::QueueBus::Util.constantize("::QueueBus::Adapters::#{class_name}").new
+        end
     end
 
-    def incoming_queue=val
-      @incoming_queue = val
-    end
+    def adapter
+      return @adapter_instance if has_adapter?
 
-    def incoming_queue
-      @incoming_queue ||= "bus_incoming"
+      raise 'no adapter has been set'
     end
 
-    def worker_middleware_stack
-      @worker_middleware_stack ||= QueueBus::Middleware::Stack.new
+    # Checks whether an adapter is set and returns true if it is.
+    def has_adapter? # rubocop:disable Naming/PredicateName
+      !@adapter_instance.nil?
     end
 
-    def hostname
-      @hostname ||= `hostname 2>&1`.strip.sub(/.local/,'')
+    def redis(&block)
+      # TODO: could allow setting for non-redis adapters
+      adapter.redis(&block)
     end
 
-    def hostname=val
-      @hostname = val
+    attr_reader :default_app_key
+    def default_app_key=(val)
+      @default_app_key = Application.normalize(val)
     end
 
-    def before_publish=(proc)
-      @before_publish_callback = proc
+    def before_publish=(callback)
+      @before_publish_callback = callback
     end
 
     def before_publish_callback(attributes)
-      if @before_publish_callback
-        @before_publish_callback.call(attributes)
-      end
-    end
-
-    def logger
-      @logger
-    end
-
-    def logger=val
-      @logger = val
+      @before_publish_callback&.call(attributes)
     end
 
     def log_application(message)
```

Comment:
@bleonard What was the intended difference between `log_application` and `log_worker`

Reply:
As far as I can tell from the usage, it looks like `log_application` is used for client applications (eg. publishing) while `log_worker` is used for logging from the driver and rider (or the processing).

I'm not sure if we need the distinction so much any more. I'm modifying queue-bus's logging so that it's controlled by setting the logger. Then the logger can be set to a level and exclude the extraneous logging.

Reply:
I was following the patterns inside resque if i recall correctly with the `puts` and `ENV` variables and such.
I made these two different because one might want to log the logistics of the application (publishing) and logistics of the workers (routing) differently.
I don't feel too strongly about it and not sure if that ever proved itself out.

Reply:
@bleonard That makes sense, right now however, if the puts are even triggered, they are sent to stdout and are actually lost within our architecture. If there is a desire to separate the two, it might be useful to just provide named configuration for `application_logger` and `worker_logger`. For now, I want to be able to capture some of these logs that we currently aren't seeing.

btw, I removed the timestamp and process ID because those are included with the default format of the ruby logger.

Reply:
Works for me, just noting history.